src/Api/Endpoints/Candidate/Visit/Dicom/Dicom.php

```diff
@@ -0,0 +1,164 @@
+    protected $visit;
+
+    /**
+     * The requested Dicom filename
+     *
+     * @var string
+     */
+    protected $tarname;
+
+    /**
+     * Contructor
+     *
+     * @param \Timepoint $visit   The requested visit
+     * @param string     $tarname The dicom filename
+     */
+    public function __construct(\Timepoint $visit, string $tarname)
+    {
+        $this->visit   = $visit;
+        $this->tarname = $tarname;
+    }
+
+    /**
+     * Return which methods are supported by this endpoint.
+     *
+     * @return array supported HTTP methods
+     */
+    protected function allowedMethods() : array
+    {
+        return array('GET');
+    }
+
+    /**
+     * Versions of the LORIS API which are supported by this
+     * endpoint.
+     *
+     * @return array a list of supported API versions.
+     */
+    protected function supportedVersions() : array
+    {
+        return array('v0.0.3-dev');
+    }
+
+    /**
+     * Handles a request to /candidates/$candid/$visit_label/dicoms/$tarname
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    public function handle(ServerRequestInterface $request) : ResponseInterface
+    {
+        $pathparts = $request->getAttribute('pathparts');
+        if (count($pathparts) === 0) {
+            switch ($request->getMethod()) {
+            case 'GET':
+                return $this->_handleGET($request);
+
+            case 'OPTIONS':
+                return (new \LORIS\Http\Response())
+                    ->withHeader('Allow', $this->allowedMethods());
+
+            default:
+                return new \LORIS\Http\Response\MethodNotAllowed(
+                    $this->allowedMethods()
+                );
+            }
+        }
+
+        // Delegate to sub-endpoints
+        return new \LORIS\Http\Response\NotFound();
+    }
+
+    /**
+     * Create an array representation of this endpoint's response body
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handleGET(ServerRequestInterface $request): ResponseInterface
+    {
+        if (!isset($this->cache)) {
+            try {
+                $dicom = $this->visit->getDicomTarByFilename($this->tarname);
+            } catch (\NotFound $e) {
+                return new \LORIS\Http\Response\NotFound();
+            }
+
+            $tarchivepath = \NDB_factory::singleton()
+                ->config()
+                ->getSetting('tarchiveLibraryDir');
+
+            $fullpath = $tarchivepath . $dicom->getArchiveLocation();
+            $info     = new \SplFileInfo($fullpath);
+            if (!$info->isFile()) {
+                return new \LORIS\Http\Response\NotFound();
+            }
+
+            if (!$info->isReadable()) {
+                return new \LORIS\Http\Response\NotFound();
```

Comment:
shouldn't it say "not readable" instead of not found here? Technically, the file is found but cannot be read due to wrong permissions. Would be helpful to send the proper error message when debugging why a file is not found.

Reply:
Same comment to apply for when we download MINC files BTW.

Reply:
@cmadjar I don't want to give to much info about the existence of files. But if you think there should be a message for unreadable file, I can add that.

Reply:
@xlecours Is there a reason why we don't want to give to much info about the existence of files? If that is the convention, no worries.

Reply:
@cmadjar I guess it's paranoia. @johnsaigle , any thoughts?

Reply:
I think it's fine in this case. The user is already logged in at this point so there's some level of trust there.